api/shape/cut.js

```diff
@@ -1,13 +1,38 @@
 import Shape from './Shape.js';
 import { cut as cutGeometry } from '@jsxcad/geometry';
-import { destructure } from './destructure.js';
+import { destructure2 } from './destructure.js';
+
+export const Cut = Shape.registerMethod('Cut', (...args) => async (shape) => {
+  const [first, rest, modes] = await destructure2(
+    shape,
+    args,
+    'geometry',
+    'geometries',
+    'modes'
+  );
+  return Shape.fromGeometry(
+    cutGeometry(
+      first,
+      rest,
+      modes.includes('open'),
```

Comment:
What do these do?...Or rather when would the user use them?

Reply:
The different between a.cut and a.Cut is if a is included implicitly in the cut, or just used as context.

a.cut(b, c) === a.Cut(self(), b, c)

One example where you might use a.Cut is a.Cut(offset(1), inset(1))

Reply:
This should be similarly true across the rest of the operations -- e.g., and, And, etc.

Reply:
That makes a lot of sense...nice!